### PR TITLE
Add designs and reorganize documents into github.io style docs

### PR DESCRIPTION
**Pull request by massand**

### CODE REVIEW

**Review by arsing — APPROVED** [review]

File: docs/keys-service.md [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+#### RSA-PKCS1
+
+```json
+{
+    "keyHandle": "string",
+    "parameters": {
+        "algorithm": "RSA-PKCS1",
+    },
+    "plaintext": "base64-encoded-string"
```
Comment (micahl) [inline docs/keys-service.md:275]: This should be "ciphertext", right?

File: docs/keys-service.md [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+
+#### RSA-NO-PADDING
+
+```json
+{
+    "keyHandle": "string",
+    "parameters": {
+        "algorithm": "RSA-NO-PADDING",
+    },
+    "plaintext": "base64-encoded-string"
```
Comment (micahl) [inline docs/keys-service.md:287]: Same as above

File: docs/keys-service.md [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+
+Only valid for ECDSA keys.
+
+Note that the request takes the message digest, ie it must be calculated by the client.
+
+```json
+{
+    "keyHandle": "string",
+    "parameters": {
+        "algorithm": "ECDSA",
```
Comment (micahl) [inline docs/keys-service.md:158]: Is there a change planned to have 'algorithm' in the 'parameters'?  It appears to currently be a sibling

File: docs/keys-service.md (hunk 1 above)
Comment (massand) [reply docs/keys-service.md]: Fixed

File: docs/keys-service.md (hunk 2 above)
Comment (massand) [reply docs/keys-service.md]: Fixed

File: docs/keys-service.md (hunk 3 above)
Comment (massand) [reply docs/keys-service.md:158]: It should not be the case - although we have been primarily using HMAC-SHA256 for IoT Hub keys. Please file an issue on this repo for this.

Comment (arsing) [reply docs/keys-service.md:158]: `algorithm` is a sibling of `parameters`. What Micah wrote is correct.

**Pull request merged**